impl/src/main/java/org/jboss/cdi/tck/tests/lookup/dynamic/handle/InstanceHandleTest.java

```diff
@@ -0,0 +1,114 @@
+package org.jboss.cdi.tck.tests.lookup.dynamic.handle;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.Bean;
+import jakarta.enterprise.inject.spi.BeanManager;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.cdi.tck.AbstractTest;
+import org.jboss.cdi.tck.shrinkwrap.WebArchiveBuilder;
+import org.jboss.cdi.tck.util.ActionSequence;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+import java.util.Iterator;
+import java.util.List;
+
+@SpecVersion(spec = "cdi", version = "4.0")
+public class InstanceHandleTest extends AbstractTest {
+
+    @Deployment
+    public static WebArchive createTestArchive() {
+        return new WebArchiveBuilder().withTestClassPackage(InstanceHandleTest.class).build();
+    }
+
+    //@SpecAssertion(section = PROGRAMMATIC_LOOKUP, id = "TODO")
+    @Test
+    public void testIsResolvable() {
+        Client client = getContextualReference(Client.class);
+        ActionSequence.reset();
+        assertNotNull(client);
+        assertTrue(client.getAlphaInstance().isResolvable());
+        assertFalse(client.getBigDecimalInstance().isResolvable());
+    }
+
+    //@SpecAssertion(section = PROGRAMMATIC_LOOKUP, id = "TODO")
+    @Test
+    public void testGetHandler() {
+        Client client = getContextualReference(Client.class);
+        BeanManager beanManager = getCurrentManager();
+
+        ActionSequence.reset();
+        assertNotNull(client);
+
+        Bean<?> alphaBean = beanManager.resolve(beanManager.getBeans(Alpha.class));
+        Instance<Alpha> instance = client.getAlphaInstance();
+
+        Instance.Handle<Alpha> alpha1 = instance.getHandle();
+        assertEquals(alphaBean, alpha1.getBean());
+        assertEquals(Dependent.class, alpha1.getBean().getScope());
+
+        String alpha2Id;
+
+        // Test try-with-resource
+        try (Instance.Handle<Alpha> alpha2 = instance.getHandle()) {
+            alpha2Id = alpha2.get().getId();
+            assertFalse(alpha1.get().getId().equals(alpha2Id));
+        }
+
+        List<String> sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(alpha2Id, sequence.get(0));
+
+        alpha1.destroy();
+        // Subsequent invocations are no-op
+        alpha1.destroy();
+
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(2, sequence.size());
+        assertEquals(alpha1.get().getId(), sequence.get(1));
```

Comment:
Here, you're calling `alpha1.get()` after `alpha1` has been `destroy()`ed, do we really guarantee that you will get the previous instance?

Reply:
This is a very good point and the behavior in the test is incorrect. Spec currently doesn't say what happens but there are basically two options:
* Throw exception if the `Handle` already resolves and subsequently destroyed the instance
* Create a new instance

The first seems like a way to go so long as we document it properly.
I will send the PRs this afternoon.

Reply:
This is now fixed in this PR.
CDI API PR is here - https://github.com/eclipse-ee4j/cdi/pull/555